dockerfiles/Dockerfile.rocm

```diff
@@ -0,0 +1,52 @@
+# --------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------
+# Dockerfile to run ONNXRuntime with MIGraphX integration
+#--------------------------------------------------------------------------
+
+FROM ubuntu:18.04
+
+ARG ONNXRUNTIME_REPO=https://github.com/Microsoft/onnxruntime
+ARG ONNXRUNTIME_BRANCH=master
+
+ENV DEBIAN_FRONTEND noninteractive
+RUN apt-get clean && apt-get update && apt-get install -y locales
+RUN locale-gen en_US.UTF-8
+RUN update-locale LANG=en_US.UTF-8
+ENV LC_ALL C.UTF-8
+ENV LANG C.UTF-8
+
+# Install rocm
+RUN apt-get update && apt-get install -y gnupg2 --no-install-recommends curl && \
+  curl -sL http://repo.radeon.com/rocm/apt/debian/rocm.gpg.key | apt-key add - && \
+  sh -c 'echo deb [arch=amd64] http://repo.radeon.com/rocm/apt/3.7/ xenial main > /etc/apt/sources.list.d/rocm.list'
```

Comment:
Should the version of ROCm be updated to ROCm 4.0?

Reply:
ROCM4.0 has a known compiler bug that will affect the MIGraphX EP, but I have no idea whether it affects the ROCM EP, that why I keep ROCM3.7 here. Hopefully, ROCM4.1 will resolve that. 

Do you still want to upgrade using ROCM4.0? If true, I can change that.

Reply:
What is the known compiler bug? ROCm 4.0 has been used for testing for ROCm EP recently.

Reply:
We have some models running incorrectly on MIGraphX when using ROCM4.0, maybe it is only related to MIGraphX. 
OK, I can use the ROCM4.0 in the docker file.